spring-vaadin/src/main/java/org/vaadin/spring/navigator/ViewProviderAccessDelegate.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2015 The original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.vaadin.spring.navigator;
+
+import com.vaadin.navigator.View;
+import com.vaadin.ui.UI;
+
+/**
+* Interface to be implemented by Spring beans that will be consulted before the Spring View provider
+* provides a view. If any of the view providers deny access, the view provider will act like no such
+* view ever existed, or show an {@link org.vaadin.spring.navigator.SpringViewProvider#setAccessDeniedViewClass(Class) access denied view}.
+*/
+public interface ViewProviderAccessDelegate {
+
+    /**
+     * Checks if the current user has access to the specified view and UI.
+     *
+     * @param beanName the bean name of the view, never {@code null}.
+     * @param ui       the UI, never {@code null}.
+     * @return true if access is granted, false if access is denied.
+     */
+    boolean isAccessGranted(String beanName, UI ui);
+
+    /**
+     * Checks if the current user has access to the specified view instance and UI. This method is invoked
+     * after {@link #isAccessGranted(com.vaadin.navigator.View, com.vaadin.ui.UI)}, when the view instance
+     * has already been created, but before it has been returned by the view provider.
+     *
+     * @param view the view instance, never {@code null}.
+     * @param ui   the UI, never {@code null}.
+     * @return true if access is granted, false if access is denied.
+     */
+    boolean isAccessGranted(String beanName, View view, UI ui);
```

Comment:
Minor detail: Javadocs missing for the beanName parameter. And maybe the order of the parameters should match the first check (view parameter last)?